podman_hpc/siteconfig.py

```diff
@@ -2,55 +2,219 @@
 import os
 import shutil
 import toml
-import yaml
+import re
+from yaml import load
+from yaml import FullLoader
 from copy import deepcopy
 from glob import glob
 
-_MOD_ENV = "PODMANHPC_MODULES_DIR"
-_HOOKS_ENV = "PODMANHPC_HOOKS_DIR"
+_ENV_PREFIX = "PODMANHPC"
+_MOD_ENV = f"{_ENV_PREFIX}_MODULES_DIR"
 _HOOKS_ANNO = "podman_hpc.hook_tool"
+_CONF_ENV = f"{_ENV_PREFIX}_CONFIG_FILE"
 
 
 class SiteConfig:
     """
     Class to represent site specific configurations for Podman-HPC.
+
+    Precedence order lowest to highest:
+    - Built-in defaults
+    - Config file
+    - Environment variables
     """
 
+    _default_conf_file = "/etc/podman_hpc/podman_hpc.yaml"
+    _valid_params = ["podman_bin", "mount_program", "modules_dir",
+                     "shared_run_exec_args", "shared_run_command",
+                     "graph_root", "run_root",
+                     "default_args", "default_run_args",
+                     "additional_stores", "hooks_dir",
+                     "localid_var", "tasks_per_node_var", "ntasks_pattern",
+                     "config_home", "mksquashfs_bin"]
+    _valid_templates = ["shared_run_args_template",
+                        "graph_root_template",
+                        "run_root_template",
+                        "default_args_template",
+                        "default_run_args_template",
+                        "additional_stores_template",
+                        "config_home_template"
+                        ]
+    _uid = os.getuid()
+    _xdg_base = f"/tmp/{_uid}_hpc"
+    config_home = f"{_xdg_base}/config"
+    run_root = _xdg_base
+    additional_stores = []
+    default_args = []
+    hooks_dir = f"{sys.prefix}/share/containers/oci/hooks.d"
+    graph_root = f"{_xdg_base}/storage"
+    squash_dir = os.environ.get(
+        "SQUASH_DIR", f'{os.environ.get("SCRATCH", "/tmp")}/storage'
+    )
+    modules_dir = "/etc/podman_hpc/modules.d"
+    shared_run_exec_args = ["-e", "SLURM_*", "-e", "PALS_*", "-e", "PMI_*"]
+    default_run_args = []
+    shared_run_command = ["sleep", "infinity"]
+    podman_bin = "podman"
+    mount_program = "fuse-overlayfs-warp"
```

Comment:
I think this is a typo and should be `fuse-overlayfs-wrap` based on what I see in `/usr/bin`

Reply:
Nevermind I see this was fixed in a later commit